libmariadb/mariadb_lib.c

```diff
@@ -3019,6 +3019,7 @@ mysql_optionsv(MYSQL *mysql,enum mysql_option option, ...)
     break;
   default:
     va_end(ap);
+    SET_CLIENT_ERROR(mysql, CR_NOT_IMPLEMENTED, SQLSTATE_UNKNOWN, 0);
     return(-1);
```

Comment:
Better to change the return code from -1 to 1. According to the documentation mysql_options/v will return non zero return code on error, so this will not break anything.